zio-http/src/main/scala/zhttp/service/client/content/handlers/DynamicContentHandler.scala

```diff
@@ -0,0 +1,39 @@
+package zhttp.service.client.content.handlers
+
+import io.netty.buffer.{ByteBuf, Unpooled}
+import io.netty.channel.{ChannelHandlerContext, SimpleChannelInboundHandler}
+import io.netty.handler.codec.http.{FullHttpResponse, HttpHeaderNames}
+import zhttp.http.{HTTP_CHARSET, Header, Status}
+import zhttp.service.Client
+import zhttp.service.Client.{ClientResponse, StringContent}
+
+final case class DynamicContentHandler() extends SimpleChannelInboundHandler[FullHttpResponse](true) {
+
+  override def channelRead0(ctx: ChannelHandlerContext, msg: FullHttpResponse): Unit = {
+    val status            = Status.fromHttpResponseStatus(msg.status())
+    val headers           = Header.parse(msg.headers())
+    val content           = Unpooled.copiedBuffer(msg.content())
+    val contentTypeHeader = headers.find(_.name == HttpHeaderNames.CONTENT_TYPE)
+
+    val response: ClientResponse = contentTypeHeader match {
+      case Some(value) =>
+        value match {
+          case Header.contentTypeTextPlain =>
+            ContentHandler.getStringContentResponse(status, headers, content)
+          case Header.contentTypeJson      =>
+            ContentHandler.getStringContentResponse(status, headers, content)
+          case _                           => ContentHandler.getStringContentResponse(status, headers, content)
+        }
+      case None        => ContentHandler.getStringContentResponse(status, headers, content)
+    }
```

Comment:
- Why are we pattern matching if in all cases we have to do the same thing?
- Decoding the content to String is expensive, and possibly unnecessary in scenarios when the content is not read by the HttpApp. Eg: An unauthenticated POST request etc.

Reply:
Stil WIP. there should be different content handling for each type of content type that we support. I used the String model only to exemplify the intent.

Reply:
refactored this part. As we do not want to have dependencies on encoders/decoders libraries ( for json, xml, etc.) this handler is basically just doing the decoding of the response and copy the content into heap. The `ClientResponse` is providing access to raw content, such that users of the client library can decode it to json/xml/ ....